constraints_tests.txt

```diff
@@ -0,0 +1,5 @@
+# This file contains package constraints needed to run the invest test suite
```

Comment:
Since the concept of a constraints file is pip-specific (I think?) but we're using conda/mamba to manage our environments, would it be worth adding a note here about the syntax that is allowed in this file?

I'm guessing [environment markers](https://peps.python.org/pep-0496/) might be a big one that wouldn't be supported without further work on our requirements conversion script, for example!